share/availability/light/sample.go

```diff
@@ -7,12 +7,31 @@ import (
 	"golang.org/x/exp/maps"
 )
 
+// SamplingResult holds the available and remaining samples.
+type SamplingResult struct {
+	Available []Sample `json:"available"`
+	Remaining []Sample `json:"remaining"`
+}
+
 // Sample represents a coordinate in a 2D data square.
 type Sample struct {
```

Comment:
I just realized that we would be better off using the `SampleIndex` defined in #3891. Ideally, having two types for the same thing should be avoided. If we end up postponing the patch release, I think we should stick to a single type.

@walldiss, @cristaloleg

Reply:
Looks like it should be resolved in second PR, that introduces second type.